arangod/Aql/ClusterNodes.cpp

```diff
@@ -131,11 +131,20 @@ DistributeNode::DistributeNode(ExecutionPlan* plan,
       _vocbase(plan->getAst()->query()->vocbase()),
       _collection(plan->getAst()->query()->collections()->get(
           base.get("collection").copyString())),
-      _varId(base.get("varId").getNumericValue<VariableId>()),
-      _alternativeVarId(base.get("alternativeVarId").getNumericValue<VariableId>()),
+      _variable(nullptr),
+      _alternativeVariable(nullptr),
       _createKeys(base.get("createKeys").getBoolean()),
       _allowKeyConversionToObject(base.get("allowKeyConversionToObject").getBoolean()),
-      _allowSpecifiedKeys(false) {}
+      _allowSpecifiedKeys(false) {
+ 
+  if (base.hasKey("variable")) {     
+    _variable = Variable::varFromVPack(plan->getAst(), base, "variable");
+    _alternativeVariable = Variable::varFromVPack(plan->getAst(), base, "alternativeVariable");
```

Comment:
what if base does not have alternativeVariable?
i would say the test should be:
`base.hasKey("variable") && base.hasKey("alternativeVariable")`
or two ifs if they can work without one another.